.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,52 @@
+name: Performance Benchmark
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  benchmark:
+    name: Run Performance Benchmarks
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [0.8, 0.9, 0.10, 0.11, 0.12, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20]
```

Comment:
this is fine for iterating on the PR, but before merging this should change to use ljharb/actions/matrix with type "majors" so it automatically includes new node majors as they come out.